components/user_settings/sidebar/user_settings_sidebar.jsx

```diff
@@ -185,39 +255,108 @@ export default class UserSettingsSidebar extends React.Component {
-                }
-                describe={this.renderAutoCloseDMLabel(this.state.settings.close_unused_direct_messages)}
-                section={'autoCloseDM'}
-                updateSection={this.updateSection}
-            />
+            <React.Fragment>
+                {contents}
+                <div className='divider-dark'/>
+            </React.Fragment>
         );
     };
 
-    renderUnreadLabel = (value) => {
-        if (value === 'true') {
-            return (
+    renderOrganizationLabel = () => {
+        const {
+            grouping,
+            unreadsAtTop,
+            favoriteAtTop,
+            sorting,
+        } = this.props.sidebarPreference;
+
+        const messages = [];
+
+        if (grouping === 'by_type') {
+            messages.push(
+                <FormattedMessage
+                    key='by_type'
+                    id='user.settings.sidebar.groupByTypeShort'
+                    defaultMessage='Group by channel type'
+                />
+            );
+        } else {
+            messages.push(
                 <FormattedMessage
-                    id='user.settings.sidebar.showUnreadSection'
-                    defaultMessage='At the top of the channel sidebar'
+                    key='none'
+                    id='user.settings.sidebar.groupByNoneShort'
+                    defaultMessage='No grouping'
                 />
             );
         }
 
-        return (
+        let sortingId;
+        let sortingDefaultMessage;
+        if (sorting === 'alpha') {
+            sortingId = 'user.settings.sidebar.sortAlphaShort';
+            sortingDefaultMessage = 'sorted alphabetically';
+        } else {
+            sortingId = 'user.settings.sidebar.sortRecentShort';
+            sortingDefaultMessage = 'sorted by recency';
+        }
+
+        messages.push(
+            <span key='comma'>{', '}</span>
+        );
+
+        messages.push(
             <FormattedMessage
-                id='user.settings.sidebar.never'
-                defaultMessage='Never'
+                key='sorting'
+                id={sortingId}
+                defaultMessage={sortingDefaultMessage}
             />
         );
+
+        let atTopId = null;
+        let atTopDefaultMessage = null;
+        if (unreadsAtTop === 'true' && favoriteAtTop === 'false') {
+            atTopId = 'user.settings.sidebar.unreadsShort';
+            atTopDefaultMessage = 'Unreads grouped separately';
+        } else if (unreadsAtTop === 'false' && favoriteAtTop === 'true') {
+            atTopId = 'user.settings.sidebar.favoritesShort';
+            atTopDefaultMessage = 'Favorites grouped separately';
+        } else if (unreadsAtTop === 'true' && favoriteAtTop === 'true') {
+            atTopId = 'user.settings.sidebar.unreadsFavoritesShort';
+            atTopDefaultMessage = 'Unreads and favorites grouped separately';
+        }
+
+        if (atTopId) {
+            messages.push(
+                <br/>
```

Comment:
You almost had keys for all of the items in these arrays, but this one is missing one
![image](https://user-images.githubusercontent.com/3277310/47752702-94bfa880-dc6b-11e8-87fe-54fd8ad4a2e7.png)